File: core/src/main/java/y2k/joyreactor/requests/UserImageRequest.java
package y2k.joyreactor.requests;

/**
 * Created by y2k on 01/10/15.
 */
public class UserImageRequest {

    private static IconStorage sStorage;

    private String name;

    public UserImageRequest(String name) {
        this.name = name;
    }

    public String execute() {
        sStorage = IconStorage.get(sStorage, "user.names", "user.icons");

        Integer id = sStorage.getImageId(name);
        return id == null ? null : "http://img0.joyreactor.cc/pics/avatar/user/" + id;
    }
}

File: core/src/main/kotlin/y2k/joyreactor/services/requests/UserImageRequest.kt
package y2k.joyreactor.services.requests

/**
 * Created by y2k on 01/10/15.
 */
class UserImageRequest {

    fun execute(name: String): String? {
        sStorage = IconStorage.get(sStorage, "user.names", "user.icons")

        val id = sStorage!!.getImageId(name)
        return if (id == null) null else "http://img0.joyreactor.cc/pics/avatar/user/" + id
    }

    companion object {

        private var sStorage: IconStorage? = null
    }
}